src/main/java/com/antelif/library/domain/service/TransactionService.java

```diff
@@ -28,9 +36,46 @@ public class TransactionService {
    * @return a transaction response DTO.
    */
   public TransactionResponse createTransaction(TransactionRequest transactionRequest) {
-    return Optional.ofNullable(transactionConverter.convertFromRequestToEntity(transactionRequest))
+    var transactionEntity = transactionConverter.convertFromRequestToEntity(transactionRequest);
+
+    validateCreation(transactionEntity);
+
+    // Change all book copy statuses to lent.
+    transactionEntity.getTransactionItems().stream()
+        .map(TransactionItemEntity::getBookCopy)
+        .forEach(BookCopyEntity::toggleStatus);
```

Comment:
Could we use the `TransactionEntity::updateTransactionItems` as used in the method below? If we could, maybe we can make toggleStatus method private.

Reply:
Changed toggle method to protected instead.